corehq/motech/repeaters/models.py

```diff
@@ -993,9 +994,11 @@ def register(self, payload, fire_synchronously=False):
             # immediately; if enabled, then it will be sent when
             # waiting_record would have been sent. Either way, this
             # change will not delay sending the payload.
-            waiting_record.delete()
             waiting_payload = DataSourceUpdate.objects.get(pk=waiting_record.payload_id)
-            doc_ids |= set(waiting_payload.doc_ids)
+            if len(waiting_payload.doc_ids) <= CASEBLOCK_CHUNKSIZE:
```

Comment:
Is it necessary to also enforce this limit when creating a payload? Or due to some upstream characteristic that I'm unaware of, is it unlikely/impossible for a payload to be larger than `CASEBLOCK_CHUNKSIZE` when first registered?

Reply:
I don't think it's necessary. Obviously this line is to prevent payloads _growing_ too big. In the past we've had problems with case imports (because we sent a payload for every doc ID) and in that case payloads are already limited to `CASEBLOCK_CHUNKSIZE` (100) cases.

Technically it's possible to submit a form that updates an arbitrary number of cases [within a request body limit](https://github.com/dimagi/commcare-hq/blob/624520d8cb1bfd0d4c366d9f9d1e83e35f9ac2ed/corehq/ex-submodules/couchforms/getters.py#L52) of `settings.MAX_UPLOAD_SIZE` (10 MB). But I think we can worry about that if/when it ever happens.